### DISCUSSION TITLE
Unexpected results when using ConfusionMatrixMetric

### DISCUSSION THREAD

**Discussion by hblanco2009** [Project-MONAI/MONAI#5966] [Q&A] [answered]
Hello guys, I'm having problems to properly compute some metrics using the Monai's ConfusionMatrixMetric class.
I guess I'm doing something wrong or there are some parameters missing. Attached you can see the example and the code. Particularly, the "accuracy" is correctly computed while the "f1 score" is not. I'm always getting the same metric value.

Any ideas on what's wrong?

Thanks in advance
H.

![expected-results](https://user-images.githubusercontent.com/61469912/217857507-fab2299c-53ee-47f4-a494-d29e5917487d.png)
![whats-wrong](https://user-images.githubusercontent.com/61469912/217857553-5ed3f88b-c29f-4e8c-86fa-d7d5829336ec.png)

**Comment by KumoLiu** [comment]
Hi @hblanco2009, if you only want to calculate the second channel, you should set `include_background` to `False`. By default, it's True.
```
metric = ConfusionMatrixMetric(metric_name="f1 score", include_background=False)
````
Hope it can help you, thx!

**Comment by yiheng-wang-nv** [comment, marked as answer]
Hi @hblanco2009 , 
The default setting for the class is that it will consider all classes, in your case the number is 2. If only count class 1, you need to set `include_background=False`.
![Screen Shot 2023-02-10 at 11 50 00 AM](https://user-images.githubusercontent.com/68361391/217996116-9aa84604-5083-49a3-891b-4dfdcb50b4b5.png)

**Reply by hblanco2009** [reply]
Thank you so much. it works perfectly now!
Best regards
H.

**Comment by giemmecci** [comment]
Hi, @KumoLiu @yiheng-wang-nv ,

Since, in some scenarios, we are interested in having the confusion matrix metrics for multiple classes (i.e., not only the positive class), what would be a way to simulate the behavior of the [classification report from scikit learn](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html) in MONAI?

For example, how can we calculate the macro average of each metric of the confusion matrix for the two classes in the example provided by the original poster? The aim would be to obtain the same information [shown in the last cell of one of your tutorials](https://github.com/Project-MONAI/tutorials/blob/main/2d_classification/mednist_tutorial.ipynb).

Thanks!

**Reply by KumoLiu** [reply]
Hi @giemmecci, thanks for your interest here.
I think you can set `reduction=mean_channel` to get the average metric for all classes.
https://github.com/Project-MONAI/MONAI/blob/00376799a2ae0e6b84fb1efe1ef9ff6d8cbedc19/monai/metrics/confusion_matrix.py#L54-L56

Hope it can help you, thanks!
